codeforamerica/vita-min
# Fyst 182 Retirement income page

## Link to pivotal/JIRA issue
- https://codeforamerica.atlassian.net/browse/FYST-182
## Is PM acceptance required?
- Yes - don't merge until JIRA issue is accepted!
## What was done?
- Adds a page to ask about retirement income
- Page is shown if fed xml indicates value for total pension amount and presence of 1099R(s)
- Checkboxes reveal inputs for amounts
- Page shows additional checkbox for spouse pensions
## How to test?
- Unit tests on controller check: show logic, spouse checkbox presence, return to review functionality
- Unit tests on form check: validations, saving values
- Section added to feature spec and corresponding fixture
- Manual test cases:
  - With fed return that has retirement income (page shows)
  - Without fed return that has retirement income (page does not show)
  - Mfj filers (spouse checkbox present)
  - Non-mfj filer (spouse checkbox absent)
  - Can't enter non-numerical value
  - Can't check box and not enter value
  - Can't enter values that add up to total over TotalTaxablePensionsAmt
## Screenshots (for visual changes)
- <img width="927" alt="image" src="https://github.com/user-attachments/assets/384c00f2-3e68-4c33-a14d-adf6291b612e">
